src/main/java/mate/academy/library/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,80 @@
+package mate.academy.library.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.library.dto.shoppingcart.CartItemDto;
+import mate.academy.library.dto.shoppingcart.CartItemRequestDto;
+import mate.academy.library.dto.shoppingcart.ShoppingCartResponseDto;
+import mate.academy.library.exception.EntityNotFoundException;
+import mate.academy.library.mapper.CartItemMapper;
+import mate.academy.library.mapper.ShoppingCartMapper;
+import mate.academy.library.model.CartItem;
+import mate.academy.library.model.ShoppingCart;
+import mate.academy.library.model.User;
+import mate.academy.library.repository.book.BookRepository;
+import mate.academy.library.repository.cartitem.CartItemRepository;
+import mate.academy.library.repository.shoppingcart.ShoppingCartRepository;
+import mate.academy.library.service.ShoppingCartService;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import java.util.Objects;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemMapper cartItemMapper;
+    private final CartItemRepository cartItemRepository;
+    private final BookRepository bookRepository;
+
+    @Override
+    public ShoppingCart save(User user) {
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        return shoppingCartRepository.save(shoppingCart);
+    }
+
+    @Override
+    public CartItemDto saveCartItem(CartItemRequestDto requestDto, User user) {
+        CartItem cartItem = cartItemMapper.toModel(requestDto);
+        cartItem.setShoppingCart(shoppingCartRepository.findShoppingCartByUser(user));

```

Comment:
It's better to split the logic, first - find book in DB, second find shopping cart, then cart.getCartItems() and check if book exists there -> increase quantity, if no ->  add book
I mean, it's better to create local variables for find results